File: java_solutions/2574.left-and-right-sum-differences.java
/*
 * @lc app=leetcode id=2574 lang=java
 *
 * [2574] Left and Right Sum Differences
 */

class Solution {
    public int[] leftRightDifference(int[] nums) {
        return IntStream.range(0, nums.length)
            .map(i -> Math.abs(IntStream.range(0, i).map(j -> nums[j]).sum() 
                             - IntStream.range(i + 1, nums.length).map(j -> nums[j]).sum()))
            .toArray();
    }
}

File: kotlin_solutions/2574.left-and-right-sum-differences.kt
/*
 * @lc app=leetcode id=2574 lang=kotlin
 *
 * [2574] Left and Right Sum Differences
 */

class Solution {fun leftRightDifference(nums: IntArray) = nums.indices.map { i -> Math.abs(nums.take(i).sum() - nums.drop(i + 1).sum()) }.toIntArray()}

